addons/mrp/tests/test_smp.py

```diff
@@ -80,3 +80,53 @@ def test_smp_produce_all_but_one(self):
         bo = mo.procurement_group_id.mrp_production_ids[-1]
         self.assertEqual(bo.backorder_sequence, count)
         self.assertEqual(bo.state, "confirmed")
+
+    def test_smp_produce_with_consumable_component(self):
+        """Create a MO for a product tracked by serial number with a consumable component.
+        Open the smp wizard, You should be able to generate all serial numbers.
+
+        BoM:
+        - 1x final product (tracked by serial number)
+        components:
+        - 2 x (storable)
+        - 4 x (consumable)
+
+        - Create a MO with 12 final products to produce.
+        - update the component quantity to 100
+        """
+        self.bom_1.product_id.uom_id = self.ref('uom.product_uom_unit')
+        self.bom_1.product_id.tracking = 'serial'
+        self.bom_1.product_qty = 1
+
+        mo_form = Form(self.env['mrp.production'])
+        mo_form.bom_id = self.bom_1
+        mo_form.product_qty = 12
+        mo = mo_form.save()
+        self.assertEqual(mo.move_raw_ids.mapped(lambda l: l.product_qty), [24, 48])
```

Comment:
add also an assert on the product type of the component in case we change bom1 in the future

Reply:
Done, Thank you!